src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -302,7 +302,8 @@ protected void CheckInitializerIfNeeded(BindingDiagnosticBag diagnostics)
             {
                 diagnostics.Add(ErrorCode.ERR_InstancePropertyInitializerInInterface, Location);
             }
-            else if (!IsAutoProperty)
+            else if (!IsAutoPropertyOrUsesFieldKeyword ||
+                ((_propertyFlags & Flags.HasAutoPropertySet) == 0) && SetMethod is { })
```

Comment:
Why can't we just check `IsAutoPropertyOrUsesFieldKeyword` here?

Reply:
This actually doesn't appear to follow what we've specified: https://github.com/dotnet/csharplang/blob/main/proposals/field-keyword.md#property-initializers.

Reply:
The proposal was specifying this inconsistently. Updated in [https://github.com/dotnet/csharplang/pull/8386](https://github.com/dotnet/csharplang/pull/8386).